ballerina/listener_utils.bal

```diff
@@ -102,6 +106,176 @@ isolated function addDefaultDirectives(__Schema schema) {
+                                string key = item[0];
+                                if fileInfo.hasKey(key) {
+                                    files[path] = <FileUpload> fileInfo[key];
+                                } else {
+                                    return "File content is missing in multipart request";
+                                }
+                            } else {
+                                return "Undefined variable found in multipart request `map`";
+                            }
+                        } else {
+                            return "Invalid type for multipart request field ‘map’ value";
+                        }
+                    }
+                } else {
+                    return "Missing multipart request field ‘map’ values";
+                }
+            } else {
+                return "Invalid type for multipart request field ‘map’";
+            }
+        }
+    } else {
+        return "Missing multipart request field ‘map’";
+    }
+    return createFileUploadInfoMap(files, variables);
+}
+
+isolated function checkAndReplaceVariablePosition(map<json> variables, string name) returns boolean {
+    //replace variable's null value with file path
+    string varName = name;
+    if name.includes(".") {
+        varName = name.substring(0, <int> name.indexOf("."));
+        string indexPart = name.substring((<int> name.indexOf(".") + 1), name.length());
+        int|error index = 'int:fromString(indexPart);
+        if variables.hasKey(varName) && variables.get(varName) is json[] && index is int {
+            json[] arrayValue = <json[]> variables.get(varName);
+            if index < arrayValue.length() {
+                arrayValue[index] = name;
+                variables[varName] = arrayValue;
+                return true;
+            }
+            return false;
+        }
+        return false;
+    }
+    return variables.hasKey(varName);
+}
+
+isolated function createFileUploadInfoMap(map<FileUpload> files, map<json> variables) returns map<FileUpload|FileUpload[]>|string {
+    map<FileUpload|FileUpload[]> fileInfo = {};
+    foreach var entry in variables.entries() {
+        string key = entry[0];
+        json value = entry[1];
+        if value is json[] {
+            FileUpload[] fileArray = [];
+            foreach json filePath in value {
+                if filePath is string {
+                    fileArray.push(<FileUpload> files[filePath]);
+                } else {
+                    return "File content is missing in multipart request";
+                }
+            }
+            fileInfo[key] = fileArray;
+        } else {
+            fileInfo[key] = <FileUpload>files[key];
+        }
+    }
+    return fileInfo;
+}
+
+isolated function forwardMultipartRequestToExecution(Engine engine, Context context, map<FileUpload|FileUpload[]> fileInfo, json payload) returns http:Response {
```

Comment:
Line too long. Please check with the style guide and set up the IDE to match those rules.